core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java

```
@@ -100,50 +106,150 @@ private static ZooCache getZooCache(Instance instance) {
+    final ArrayList<Namespace.ID> singleId = new ArrayList<>(1);
+    populateMap(instance, (id, name) -> {
+      if (name.equals(namespaceName))
+        singleId.add(new Namespace.ID(id));
+    });
+    if (singleId.isEmpty())
+      throw new NamespaceNotFoundException(null, namespaceName, "getNamespaceId() failed to find namespace");
+    return singleId.get(0);
+  }
+
+  /**
+   * Look for namespace ID in ZK. Fail quietly by logging and returning null.
+   */
+  public static Namespace.ID lookupNamespaceId(Instance instance, String namespaceName) {
+    Namespace.ID id = null;
+    try {
+      id = getNamespaceId(instance, namespaceName);
+    } catch (NamespaceNotFoundException e) {
+      if (log.isDebugEnabled())
+        log.debug("Failed to find namespace ID from name: " + namespaceName, e);
+    }
+    return id;
+  }
+
+  /**
+   * Return true if namespace name exists
+   */
+  public static boolean namespaceNameExists(Instance instance, String namespaceName) {
+    return lookupNamespaceId(instance, namespaceName) != null;
+  }
+
+  /**
+   * Look for namespace name in ZK. Throw NamespaceNotFoundException if not found.
+   */
+  public static String getNamespaceName(Instance instance, Namespace.ID namespaceId) throws NamespaceNotFoundException {
+    final ArrayList<String> singleName = new ArrayList<>(1);
+    populateMap(instance, (id, name) -> {
+      if (id.equals(namespaceId.canonicalID()))
+        singleName.add(name);
+    });
+    if (singleName.isEmpty())
+      throw new NamespaceNotFoundException(namespaceId.canonicalID(), null, "getNamespaceName() failed to find namespace");
+    return singleName.get(0);
+  }
+
+  /**
+   * Look for namespace name in ZK. Fail quietly by logging and returning null.
+   */
+  public static String lookupNamespaceName(Instance instance, Namespace.ID namespaceId) {
+    String name = null;
+    try {
+      getNamespaceName(instance, namespaceId);
```

Why suppress this exception? It seems like something that should propagate to the caller.